From d8092a183cb60d7f272bd3e48843ad6e46becb75 Mon Sep 17 00:00:00 2001
From: Carlos Munoz <carlos.munoz@caviumnetworks.com>
Date: Tue, 2 Jul 2013 13:48:11 -0700
Subject: [PATCH 167/518] netdev: octeon-ethernet: Configure the PHY interface
 type correctly.

Source: Cavium Networks, Inc.
MR: 00000
Type: Integration
Disposition: Merged from Octeon Tree
ChangeID: 62a22e93ac43718ff0b460300acb9ae04dbdfc7b
Description:

Signed-off-by: Carlos Munoz <carlos.munoz@caviumnetworks.com>
Signed-off-by: Corey Minyard <cminyard@mvista.com>
[Original patch taken from Cavium SDK 3.1.1 525]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/net/ethernet/octeon/ethernet-mdio.c | 18 +++++++++++++++++-
 1 file changed, 17 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/octeon/ethernet-mdio.c b/drivers/net/ethernet/octeon/ethernet-mdio.c
index bbb6452..661c6e8 100644
--- a/drivers/net/ethernet/octeon/ethernet-mdio.c
+++ b/drivers/net/ethernet/octeon/ethernet-mdio.c
@@ -387,6 +387,7 @@ int cvm_oct_phy_setup_device(struct net_device *dev)
 {
 	struct octeon_ethernet *priv = netdev_priv(dev);
 	struct device_node *phy_node;
+	phy_interface_t iface;
 
 	if (!priv->of_node)
 		goto no_phy;
@@ -395,8 +396,23 @@ int cvm_oct_phy_setup_device(struct net_device *dev)
 	if (!phy_node)
 		goto no_phy;
 
+	switch (priv->imode) {
+	case CVMX_HELPER_INTERFACE_MODE_GMII:
+		iface = PHY_INTERFACE_MODE_GMII;
+		break;
+	case CVMX_HELPER_INTERFACE_MODE_RGMII:
+		iface = PHY_INTERFACE_MODE_RGMII;
+		break;
+	case CVMX_HELPER_INTERFACE_MODE_SGMII:
+		iface = PHY_INTERFACE_MODE_SGMII;
+		break;
+	default:
+		iface = PHY_INTERFACE_MODE_NA;
+		break;
+	}
+
 	priv->phydev = of_phy_connect(dev, phy_node, cvm_oct_adjust_link, 0,
-				      PHY_INTERFACE_MODE_GMII);
+				      iface);
 
 	if (priv->phydev == NULL)
 		return -ENODEV;
-- 
1.9.1

